lib/FreeRTOS/freertos-plus-mqtt/agent_command_pool.c

```diff
@@ -62,83 +54,120 @@ struct AgentMessageContext
 static Command_t commandStructurePool[ MQTT_COMMAND_CONTEXTS_POOL_SIZE ];
 
 /**
- * @brief A queue used to guard the pool of Command_t structures. Structures
- * may be obtained by receiving a pointer from the queue, and returned by
- * sending the pointer back into it.
+ * @brief An array of semaphores to guard each Command_t structure in the pool.
+ * Structures must be obtained by first obtaining its associated semaphore.
  */
-static AgentMessageContext_t commandStructQueue;
+static SemaphoreHandle_t commandSems[ MQTT_COMMAND_CONTEXTS_POOL_SIZE ];
 
 /**
  * @brief Initialization status of the queue.
```

Comment:
This comment needs updating: _Initialization status of the command semaphore array._